code/modules/mob/living/basic/space_fauna/demon/demon_subtypes.dm

```diff
@@ -0,0 +1,163 @@
+		return
+
+	face_atom(attack_target)
+
+	var/mob/living/victim = attack_target
+	victim.take_bodypart_damage(brute = 20, wound_bonus = wound_bonus) // don't worry, there's more punishment when they hit something
+
+	visible_message(
+		span_danger("[src] slams into [victim] with monstrous strength!"),
+		span_danger("You slam into [victim] with monstrous strength!"),
+		ignored_mobs = victim,
+	)
+	to_chat(victim, span_userdanger("[src] slams into you with monstrous strength, sending you flying like a ragdoll!"))
+
+	var/turf/yeet_target = get_edge_target_turf(victim, dir)
+	victim.throw_at(yeet_target, 10, 5, src)
+	COOLDOWN_START(src, slam_cooldown, slam_cooldown_time)
+	log_combat(src, victim, "slaughter slammed")
+
+/// Proc that we execute on attacking someone to keep track of our hitstreaks and wound bonuses. Also handles triggering the bodyslam on attacks.
+/mob/living/basic/demon/slaughter/proc/on_attack(mob/living/source, atom/attack_target, proximity_flag, list/modifiers)
+	SIGNAL_HANDLER
+
+	if(LAZYACCESS(modifiers, RIGHT_CLICK))
+		bodyslam(attack_target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!iscarbon(attack_target))
+		return
+
+	var/mob/living/carbon/target = attack_target
+	if(target.stat == DEAD || isnull(target.mind) || (current_hitstreak > wound_bonus_hitstreak_max))
+		return
+
+	current_hitstreak++
+	wound_bonus += wound_bonus_per_hit
+	bare_wound_bonus += wound_bonus_per_hit
+
+/// The laughter demon! It's everyone's best friend! It just wants to hug them so much, it wants to hug everyone at once!
+/mob/living/basic/demon/slaughter/laughter
+	name = "laughter demon"
+	real_name = "laughter demon"
+	desc = "A large, adorable creature covered in armor with pink bows."
+	speak_emote = list("giggles", "titters", "chuckles", "guffaws", "laughs")
+	response_help_continuous = "hugs"
+	attack_verb_continuous = "wildly tickles"
+	attack_verb_simple = "wildly tickle"
+
+	attack_sound = 'sound/items/bikehorn.ogg'
+	attack_vis_effect = null
+	death_sound = 'sound/misc/sadtrombone.ogg'
+
+	icon_state = "bow_demon"
+	icon_living = "bow_demon"
+	death_message = "fades out, as all of its friends are released from its prison of hugs."
+
+	antag_type = /datum/antagonist/slaughter/laughter
+	crawl_type = /datum/action/cooldown/spell/jaunt/bloodcrawl/slaughter_demon/funny
+
+/mob/living/basic/demon/slaughter/laughter/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_LIVING_DEATH, PROC_REF(on_death))
+	if(check_holidays(APRIL_FOOLS))
+		icon_state = "honk_demon"
+
+/mob/living/basic/demon/slaughter/laughter/grant_loot()
+	return list() // we do our thing... with the kittens...
+
+/// We do our own special thing on death, which is to spawn a kitten.
+/mob/living/basic/demon/slaughter/laughter/proc/on_death()
+	SIGNAL_HANDLER
+	var/mob/living/simple_animal/pet/cat/kitten/kitty = new(drop_location())
+	kitty.name = "Laughter"
+
+/mob/living/basic/demon/slaughter/laughter/ex_act(severity)
```

Comment:
I'm sure it was true of the base implementation but why is this different for laughter demons?

Reply:
it was introduced in a task-specific PR. #19005. no clue why it doesn't extend to slaughter demons, maybe it's because laughter demons don't permanently remove the player from the round?